### PR TITLE
fix: log a warning on invalid schema validation

### PR DESCRIPTION
**Pull request by brettstack**
*Issue #, if available:*

*Description of changes:*
Log a warning on invalid schema validation.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samtranslator/parser/parser.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 from samtranslator.validator.validator import SamTemplateValidator
 from samtranslator.model import ResourceTypeResolver, sam_resources
 from samtranslator.plugins import LifeCycleEvents
+import logging
```
Comment (jfuss) [inline samtranslator/parser/parser.py]: This uses the root logger, which will log in any one using SAM to translate without a way to turn it off. 

logging should be done by adding a `__name__`.

`logger = logging.getLogger(__name__)` and then log by doing `logger.warning("Logging a Warning")`

Comment (brettstack) [reply samtranslator/parser/parser.py]: Can you send a PR? How do other consumers then turn that off?

Comment (jfuss) [reply samtranslator/parser/parser.py]: https://stackoverflow.com/a/35325280

Should magically work. You can also manipulate the libraries logger too. For example, SAM CLI could then turn the SAM Translators logging on for a given level if they so wanted to.

Comment (fatbasstard) [reply samtranslator/parser/parser.py]: Based on https://stackoverflow.com/questions/24750523/disable-output-of-root-logger we're handling it like this now:

https://github.com/awslabs/cfn-python-lint/pull/117/files#diff-6d7631d1ad3e87ab0c24664abb771bc4R79

**Pull request merged**